contrib_bots/lib/poll.py

```diff
@@ -0,0 +1,105 @@
+
+
+class PollHandler(object):
+    '''
+    This plugin facilitates polling.
+    '''
+
+    def __init__(self):
+        # polls is a dict of dict. First dict corresponds to poll title,
+        # second refers to answers in poll
+        self.polls = dict()
+
+        # streams is a convenience object that keeps track of the latest
+        # question for each stream.
+        self.streams = dict()
```

Comment:
Sorry, my comment was a bit unclear - I meant the archive option to get the polls that are closed :) 

The limit in the poll list sounds sensible, so go for it :) Adding an option for getting a certain number of polls makes sense - I'm wondering how many polls could a Zulip topic have, since they are usually rather short-lived, but limiting here is still a good idea.

I'm commenting from my phone, so no backticks or other edition options - sorry for that, wanted to get back to you ASAP so I'm not blocking your work.